# Nike Shoes Sales Data Analysis  
### Problem Statement
The purpose of this data analysis is to check the data, and conduct data preparation and cleaning.

In [1]:
import pandas as pd

In [3]:
#load CSV in a DataFrame.
df = pd.read_csv('nike_shoes_sales.csv')

In [5]:
#get information about the CSV file
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 643 entries, 0 to 642
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   product_name   643 non-null    object 
 1   product_id     643 non-null    object 
 2   listing_price  643 non-null    int64  
 3   sale_price     643 non-null    int64  
 4   discount       643 non-null    int64  
 5   brand          643 non-null    object 
 6   description    640 non-null    object 
 7   rating         643 non-null    float64
 8   reviews        643 non-null    int64  
 9   images         572 non-null    object 
dtypes: float64(1), int64(4), object(5)
memory usage: 50.4+ KB


In [4]:
#display first 5 rows
df.head(5)

product_name  product_id  listing_price  sale_price  \
0  Nike Air Force 1 '07 Essential  CJ1646-600              0        7495   
1            Nike Air Force 1 '07  CT4328-101              0        7495   
2    Nike Air Force 1 Sage Low LX  CI3482-200              0        9995   
3             Nike Air Max Dia SE  CD0479-200              0        9995   
4             Nike Air Max Verona  CZ6156-101              0        9995   

   discount brand                                        description  rating  \
0         0  Nike  Let your shoe game shimmer in the Nike Air For...     0.0   
1         0  Nike  The legend lives on in the Nike Air Force 1 '0...     0.0   
2         0  Nike  Taking both height and craft to new levels, th...     0.0   
3         0  Nike  Designed for a woman's foot, the Nike Air Max ...     0.0   
4         0  Nike  Pass on the good vibes in the Nike Air Max Ver...     0.0   

   reviews                                             images  
0        0  ["https://static.nike.com/a/images/t_PDP_1728_...  
1        0  ["https://static.nike.com/a/images/t_PDP_1728_...  
2        0  ["https://static.nike.com/a/images/t_PDP_1728_...  
3        0  ["https://static.nike.com/a/images/t_PDP_1728_...  
4        0  ["https://static.nike.com/a/images/t_PDP_1728_...

According to df.info, there are 643 records, however some rows to not have description and image URL  
No description - 3 records  
No image - 71 records

In [8]:
descnan = df[df.description.isnull()]
print(descnan)

                        product_name  product_id  listing_price  sale_price  \
155  Nike React Infinity Run Flyknit  CD4371-001              0       13995   
263  Nike React Infinity Run Flyknit  CD4372-004              0       13995   
508             Nike Free X Metcon 2  CD8526-446              0        9995   

     discount brand description  rating  reviews  \
155         0  Nike         NaN     4.6       68   
263         0  Nike         NaN     4.2       29   
508         0  Nike         NaN     3.0        7   

                                                images  
155  ["https://static.nike.com/a/images/t_PDP_1728_...  
263  ["https://static.nike.com/a/images/t_PDP_1728_...  
508  ["https://static.nike.com/a/images/t_PDP_1728_...  


In [10]:
imagenan = df[df.images.isnull()]
print(imagenan)

                             product_name  product_id  listing_price  \
7                      Nike Free Metcon 3  CJ0861-017              0   
21                     Nike SB Bruin High  923112-001           7995   
30                           Jordan Break  AR6374-402              0   
36               Nike Air Force 1 '07 LV8  CD0888-002              0   
38                       Nike Air Max Dia  CJ0636-100              0   
..                                    ...         ...            ...   
606                      Nike Air Max 720  AR9293-603          15995   
622               Nike Air Force 1 '07 SE  CI3446-001              0   
634    Nike Mercurial Vapor 13 Academy IC  AT7993-414           7495   
639            Nike Phantom Venom Club IC  AO0578-717           4995   
640  Nike Mercurial Superfly 7 Academy TF  AT7978-414           8495   

     sale_price  discount brand  \
7          9995         0  Nike   
21         5597         0  Nike   
30         1995         0  Nik